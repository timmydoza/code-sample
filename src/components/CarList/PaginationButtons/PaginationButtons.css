.paginationButtons {
  text-align: center;
  padding: 1em;
}

.paginationButtons button {
  margin: 0 2em;
}