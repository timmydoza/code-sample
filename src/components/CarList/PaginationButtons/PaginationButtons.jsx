import React from 'react';
import PropTypes from 'prop-types';
import styles from './PaginationButtons.css';

const PaginationButtons = (props) => {
  if (props.totalPages < 2) {
    return null;
  }

  return (
    <div className={styles.paginationButtons}>
      <button onClick={() => props.setPage(-1)}>&lt;</button>
      <span>{props.currentPage} of {props.totalPages}</span>
      <button onClick={() => props.setPage(1)}>&gt;</button>
    </div>
  );
};

PaginationButtons.propTypes = {
  setPage: PropTypes.func.isRequired,
  currentPage: PropTypes.number,
  totalPages: PropTypes.number,
};

export default PaginationButtons;
