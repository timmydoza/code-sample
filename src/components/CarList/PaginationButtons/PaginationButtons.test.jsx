import React from 'react';
import renderer from 'react-test-renderer';
import PaginationButtons from './PaginationButtons';

describe('the PaginationButtons component', () => {
  let paginationButtons, setPage;

  beforeEach(() => {
    setPage = jest.fn();
    paginationButtons = renderer
      .create(<PaginationButtons currentPage="4" totalPages="11" setPage={setPage}/>)
      .toJSON()
  });

  it('renders correctly', () => {
    expect(paginationButtons).toMatchSnapshot();
  });

  it('has working buttons', () => {
    paginationButtons.children[0].props.onClick();
    paginationButtons.children[2].props.onClick(); //Click the buttons
    expect(setPage.mock.calls).toEqual([[-1],[1]]);
  })

});