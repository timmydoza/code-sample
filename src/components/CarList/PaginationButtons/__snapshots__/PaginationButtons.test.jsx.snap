// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PaginationButtons component renders correctly 1`] = `
<div
  className="paginationButtons"
>
  <button
    onClick={[Function]}
  >
    &lt;
  </button>
  <span>
    4
     of 
    11
  </span>
  <button
    onClick={[Function]}
  >
    &gt;
  </button>
</div>
`;
